src/Tools/CompiledBlock.extension.st

```diff
@@ -0,0 +1,20 @@
+Extension { #name : 'CompiledBlock' }
+
+{ #category : '*Tools' }
+CompiledBlock >> browse [
+	| method |
+
+	method := self method.
+	
+	method isInstalled
+		ifTrue: [
+			^ Smalltalk tools browser
+				openOnClass: self methodClass
+				selector: self selector
+				highlight: self sourceNode sourceCode ]
+		ifFalse: [
+			^ Smalltalk tools browser
+				openOnClass: self methodClass
+				selector: self selector ]
```

Comment:
I like the idea, but beware of edge cases that can lead to strange behavior.
If the method does not exist, it opens on the class.
If the block comes from the playground, it opens on the UndefinedObject class.
From an inspector script, it opens on the class of the object being inspected.
While these cases make some sense because they are the  "location" where the block came from, no methods are displayed, so this can feel odd (like "how did I end up on the XYZ class when browsing a block?")

Reply:
Agreed. I don't really know what _should_ be done for those edgecases. The current code will behave similarly to `CompiledMethod>>#browse` (that's intentional).

I think we have the following cases:

- Method is installed: Open the browser on that method and highlight the block
- A method with the same name is installed and contains a block with the same code: Open the browser on that method. Don't highlight anything **(here we might want to highlight the code 🤔)**
- A method with the same name is installed but does not contain a similar block: Open the browser on that method. Don't highlight anything.
- Method is not installed, class is installed: **[Currently]** Open the browser on that class
- Method is not installed, a class with the same name is installed: _Don't know what happens, it should be the same as with `CompiledMethod>>#browse`_
- Method is not installed, class is not installed: _Don't know what happens, it should be the same as with `CompiledMethod>>#browse`_

Reply:
A big difference with CompiledMethods is that they are usually installed in a class, unless we are working at a meta level.
Blocks appear much more often, and they are never installed when we work in a "scripting" context (playground, inspector, when we modify and execute code without saving (debugger, code browser...), etc.).

If the block is installed, browsing the code where it is defined sounds like a good idea.
If not, none of the options seem satisfactory to me.
Also, I personally always interpreted the "browse" action as "browse the class of the object", so I'm already thrown off when "browsing" methods, packages, and classes.

I'm not sure if it's possible to highlight the block when opening the code browser...?
I know it's supported, for example when using `Find & Replace`, but I've never seen Calypso opened with code pre-highlighted.